fearless/Common/Protocols/SheetAlertPresentable.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import UIKit
+
+struct SheetAlertPresentableAction {
+    let title: String
+    let style: TriangularedButton
+
+    let handler: (() -> Void)?
+}
+
+struct SheetAlertPresentableViewModel {
+    let title: String
+    let titleStyle: SheetAlertPresentableStyle
+    let subtitle: String?
+    let subtitleStyle: SheetAlertPresentableStyle?
+    let actions: [SheetAlertPresentableAction]
+}
+
+protocol SheetAlertPresentable {
+    func present(
+        viewModel: SheetAlertPresentableViewModel,
+        from view: ControllerBackedProtocol?
+    )
+}
+
+extension SheetAlertPresentable {
+    func present(
+        viewModel: SheetAlertPresentableViewModel,
+        from view: ControllerBackedProtocol?
+    ) {
+        var currentController = view?.controller
+
+        if currentController == nil {
+            currentController = UIApplication.shared.delegate?.window??.rootViewController
```

Comment:
it looks like you wanted to present this modal universally. best way to do it is to create new UIWindow and present modal here (it's apple recommendation for presenting alerts)